pkg/cli/image/extract/extract.go

```diff
@@ -375,17 +386,33 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
-				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
 				if err != nil {
-					if imagemanifest.IsImageForbidden(err) {
-						msg := fmt.Sprintf("image %q does not exist or you don't have permission to access the repository", from)
-						return imagemanifest.NewImageForbidden(msg, err)
-					}
-					if imagemanifest.IsImageNotFound(err) {
-						msg := fmt.Sprintf("image %q not found: %s", from, err.Error())
-						return imagemanifest.NewImageNotFound(msg, err)
+					if strings.Contains(err.Error(), "a manifest image could not be found for this platform") {
+						// Translate the current runtime OS to linux when looking through a manifest since there will not be a darwin/'archType' in a manifest image.
+						// In cases where the runtime OS is windows or darwin (i.e. macOS), there is no expectation of a manifest image with those particular OS's.
+						if o.FilterOptions.DefaultOSFilter {
+							runtimeOS := runtime.GOOS
+							arch := runtime.GOARCH
+							if linuxConversionWarning == "" {
+								linuxConversionWarning = fmt.Sprintf("Warning: a manifest image could not be found for this platform: %s/%s\n", runtimeOS, arch) +
+									fmt.Sprintf("Warning: converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image\n", runtimeOS, arch)
+							}
+							o.FilterOptions.OSFilter, err = regexp.Compile("linux/" + arch)
+							o.FilterOptions.FilterByOS = "linux/" + arch
+
+							if err != nil {
+								return fmt.Errorf("failed to compile OSFilter for linux/%s: %v", arch, err)
+							}
+
+							srcManifest, location, err = retrieveSourceManifest(ctx, from, repo, o)
```

Comment:
I see that the main reason of `retrieveSourceManifest` function is being called also in here. I still think, we might need this function